zapetw/core_windows.go

```diff
@@ -0,0 +1,89 @@
+package zapetw
+
+import (
+	"github.com/Microsoft/go-winio/pkg/etw"
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+// <organization_name>-<product_name>-<component_name>
+const providername = "Azure-CNI-CNS-Monitoring"
+
+type ETWCore struct {
+	provider  *etw.Provider
+	eventName string
+	encoder   zapcore.Encoder
+	fields    []zapcore.Field
+	zapcore.LevelEnabler
+}
+
+func NewETWCore(eventName string, encoder zapcore.Encoder, levelEnabler zapcore.LevelEnabler) (*ETWCore, error) {
+	provider, err := etw.NewProviderWithOptions(providername)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create ETW provider")
+	}
+	return &ETWCore{
+		provider:     provider,
+		eventName:    eventName,
+		encoder:      encoder,
+		LevelEnabler: levelEnabler,
+	}, nil
+}
+
+func (core *ETWCore) With(fields []zapcore.Field) zapcore.Core {
+	return &ETWCore{
+		provider:     core.provider,
```

Comment:
in geneva/kusto, do we have fields to determine where this log is coming from like VM, subscription id, region,etc?

Reply:
yes lets add that including resource group(rg). VM names are not unique. subid+rg+role_instance uniquely identifies the VM